src/ExperimentalCycles.mo

```diff
@@ -0,0 +1,60 @@
+/// Cycles
+
+/// Usage of the Internet Computer is measured, and paid for, in _cycles_.
+/// This library provides imperative operations for observing cycles, transferring cycles and
+/// observing refunds of cycles.
+
+
+/// **WARNING:** This low-level API is **experimental** and likely to change or even disappear.
+/// Dedicated syntactic support for manipulating cycles may be added to the language in future, obsoleting this library.
+///
+/// **NOTE:** Since cycles measure computational resources, the value of
+/// `balance()` can change from one call to the next.
+
+import Prim "mo:prim";
+module {
+
+  /// Returns the actor's current balance of cycles as `amount`.
+  public func balance() : (amount : Nat) {
+    Prim.nat64ToNat(Prim.cyclesBalance())
+  };
+
+  /// Returns the currently available `amount` of cycles.
+  /// The amount available is the amount received in the current call,
+  /// minus the cumulative amount `accept`ed by this call.
+  /// On exit from the current shared function or async expression via `return` or `throw`
+  /// any remaining available amount is automatically
+  /// refunded to the caller/context.
+  public func available() : (amount : Nat) {
+    Prim.nat64ToNat(Prim.cyclesAvailable())
+  };
+
+  /// Transfers `amount` from `available()` to `balance()`, returning `remainder` available.
+  /// Traps if trying to accept more cycles than are available.
+  public func accept(amount : Nat) : (remainder : Nat) {
+    Prim.nat64ToNat(Prim.cyclesAccept(Prim.natToNat64(amount)));
+  };
```

Comment:
```suggestion
  /// Transfers up to `amount` from `available()` to `balance()`
  /// Returns the amount actually transferred, which may be less than 
  /// requested, e.g. if less is available, or if canister balance limits are reached
  public func accept(amount : Nat) : (accepted : Nat) {
    Prim.nat64ToNat(Prim.cyclesAccept(Prim.natToNat64(amount)));
  };
```

Reply:
Oh dear, must not have read the spec carefully enough.